server/src/main/java/org/candlepin/resource/ActivationKeyResource.java

```diff
@@ -227,8 +227,10 @@ public ActivationKeyDTO addPoolToKey(
         ActivationKey key = this.fetchActivationKey(activationKeyId);
         Pool pool = findPool(poolId);
 
-        // Throws a BadRequestException if adding pool to key is a bad idea
-        activationKeyRules.validatePoolForActKey(key, pool, quantity);
+        activationKeyRules.validatePoolForActivationKey(key, pool, quantity)
```

Comment:
This is a very interesting way of handling this. I'm not quite sure how to feel about it yet, so let's leave it as is and see if it establishes a pattern for general output processing of the Java rules output. If so, keep them all; otherwise I would argue this is probably an expensive way of doing a null check.

Reply:
Ondrej suggested it initially. After discussing with him again, we decided to revert it, since it doesn't gain us something in this case, while introducing a performance hit. And I wouldn't want for it to become a habit throughout the rest of the "move rules to Java".